### PR TITLE
Lightning filter fixes

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Indexed id fields are incorrectly included as (unindexed) subfilters in lightning mode

Also fixes loading state with boolean lightning filters.

Note that indexed ObjectId fields offer indexed matches/querying but do not offer ixscan substring prefix search, so search results are not enabled. Only exact 24 hex character searches are allowed

### Before

https://github.com/user-attachments/assets/f2bb1301-ddc2-4917-957b-5fab6c779298

### After

https://github.com/user-attachments/assets/ae613edc-7804-47d9-9e71-2122700acc53

## How is this patch tested? If it is not, please explain why.

Selector test

## Release Notes

* Added indexed `id` fields to lightning filters

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Enhanced loading indicator logic in the Filters component for improved user feedback during data loading.
	- Updated filtering logic in the lightning module for more dynamic path validation.

- **Bug Fixes**
	- Adjusted the FILTERS object key definition to ensure proper referencing.

- **Tests**
	- Introduced unit tests for the lightning module to validate selector functionality and improve test coverage.

### CODE REVIEW

File: app/packages/core/src/components/Filters/StringFilter/Checkboxes.tsx [hunk 1]
```diff
@@ -180,7 +180,7 @@ const Checkboxes = ({
   const show = useRecoilValue(showSearchSelector({ modal, path }));
   const getCount = useGetCount(modal, path);
 
-  if (loading) {
+  if (loading || (!show && results === null)) {
```
Comment (minhtuev) [inline app/packages/core/src/components/Filters/StringFilter/Checkboxes.tsx:183]: probably we should add a comment here to explain this

**Review by minhtuev — APPROVED** [review]
Added a comment otherwise LGTM

**Comment by findtopher** [comment]
If this is fixes, should this target release/v0.25.1?

**Comment by benjaminpkane** [comment]
Could use more testing. Converting draft

**Pull request merged**